word2vec, adds MOJO support to the word2vec algorithm

michalkurka opened the pull request.

michalkurka commented on h2o-algos/src/main/java/hex/word2vec/Word2VecMojoWriter.java:
```diff
@@ -0,0 +1,45 @@
+    // Vocabulary
+    startWritingTextFile("vocabulary");
+    for (BufferedString word : model._output._words) {
+      writeln(word.toString()); // FIXME: escape new lines
```
@st-pasha do we have a convenient function for escaping of new line characters?

string can already contain "\n", eg.:

"this\none
string"

should be escaped to "this\\none\nstring"

st-pasha replied:
perhaps `word.replaceAll("\n", "\\n")` ?

michalkurka replied:
I don't think this is enough. If I escape "\n" I also need to escape "\\" otherwise I cannot tell from a "\\n" that was in the String and "\\n" that I created by escape a new line character

michalkurka commented:
Thanks, backwards compatibility tests are definitely a good suggestion

michalkurka commented:
@st-pasha I added one more commit. It handles escaping of the newline character. I changed some functions in ModelMojoWriter & ModelMojoReader, please let me know if you are okay with the changes.
